instrumentation/aws-lambda-1.0/library/build.gradle.kts

```diff
@@ -9,21 +9,24 @@ dependencies {
   compileOnly("com.google.auto.value:auto-value-annotations")
   annotationProcessor("com.google.auto.value:auto-value")
 
-  library("com.amazonaws:aws-lambda-java-core:1.0.0")
+  library("com.amazonaws:aws-lambda-java-core:1.2.1")
```

Comment:
Usually the name of the module (in this case `aws-lambda-1.0`) includes the earliest supported library version. Are you sure that it still works with 1.0? If it does consider not changing this version.

Reply:
This artefact is used to deploy the AWS lambda layer. By using the latest version you have access to more feature of the lambda context: https://github.com/aws/aws-lambda-java-libs/blob/master/aws-lambda-java-core/RELEASE.CHANGELOG.md. I see nothing to gain from leaving the old version in, but it's out of scope for this PR and I don't want it to block it, so let me know if you want me to roll it back.

Reply:
> By using the latest version you have access to more feature of the lambda context

The thing is, we don't necessarily want to use the newer features - the javaagent instrumentation of AWS lambdas captures references to used library code at compile time and verifies that they're present in the runtime (read [muzzle docs](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/muzzle.md) for more info on that); if we instrument a lambda that is compiled against 1.0 and doesn't have the methods we use in the instrumentation, the whole instrumentation will be turned off to prevent linking errors.

Reply:
Reverted.